pkg/rules/signature/signature.go

```diff
@@ -61,97 +63,109 @@ func Find(target string, partialEval bool, rulesDir string, rules []string, aioE
+	capabilities.GetInstance().Requested(
+		func() error {
+			err := filepath.WalkDir(dir,
+				func(path string, d fs.DirEntry, err error) error {
+					if err != nil {
+						return err
+					}
+
+					if d.IsDir() || filepath.Ext(d.Name()) != ".so" {
+						return nil
+					}
+
+					p, err := plugin.Open(path)
+					if err != nil {
+						logger.Error("opening plugin " + path + ": " + err.Error())
+						return err
+					}
+					export, err := p.Lookup("ExportedSignatures")
+					if err != nil {
+						logger.Error("missing Export symbol in plugin " + d.Name())
+						return err
+					}
+					sigs := *export.(*[]detect.Signature)
+					res = append(res, sigs...)
+					return nil
+				})
 			return err
-		}
-
-		if d.IsDir() || filepath.Ext(d.Name()) != ".so" {
-			return nil
-		}
+		},
+		cap.DAC_OVERRIDE,
+	)
 
-		p, err := plugin.Open(path)
-		if err != nil {
-			logger.Error("opening plugin " + path + ": " + err.Error())
-			return err
-		}
-		export, err := p.Lookup("ExportedSignatures")
-		if err != nil {
-			logger.Error("missing Export symbol in plugin " + d.Name())
-			return err
-		}
-		sigs := *export.(*[]detect.Signature)
-		res = append(res, sigs...)
-		return nil
-	})
 	return res, nil
 }
 
 func findRegoSigs(target string, partialEval bool, dir string, aioEnabled bool) ([]detect.Signature, error) {
+	var res []detect.Signature
+
 	modules := make(map[string]string)
 	modules["helper.rego"] = embedded.RegoHelpersCode
 
 	regoHelpers := []string{embedded.RegoHelpersCode}
-	filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
-		if err != nil {
-			return err
-		}
-
-		if d.IsDir() || d.Name() == "helpers.rego" {
-			return nil
-		}
-
-		if !isHelper(d.Name()) {
-			return nil
-		}
-
-		helperCode, err := ioutil.ReadFile(path)
-		if err != nil {
-			logger.Error("reading file " + path + ": " + err.Error())
-			return nil
-		}
-
-		regoHelpers = append(regoHelpers, string(helperCode))
-		modules[path] = string(helperCode)
-		return nil
-	})
-
-	var res []detect.Signature
-	filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
-		if err != nil {
-			return err
-		}
-
-		if d.IsDir() || !isRegoFile(d.Name()) || isHelper(d.Name()) {
-			return nil
-		}
 
-		regoCode, err := ioutil.ReadFile(path)
-		if err != nil {
-			logger.Error("reading file " + path + ": " + err.Error())
-			return nil
-		}
-		modules[path] = string(regoCode)
-		if aioEnabled {
-			return nil
-		}
-		sig, err := regosig.NewRegoSignature(target, partialEval, append(regoHelpers, string(regoCode))...)
-		if err != nil {
-			newlineOffset := bytes.Index(regoCode, []byte("\n"))
-			if newlineOffset == -1 {
-				codeLength := len(regoCode)
-				if codeLength < 22 {
-					newlineOffset = codeLength
-				} else {
-					newlineOffset = 22
+	capabilities.GetInstance().Requested(
+		func() error {
+			filepath.WalkDir(dir,
+				func(path string, d fs.DirEntry, err error) error {
+					if err != nil {
+						return err
```

Comment:
Maybe add an error log here as well?

Reply:
Won't hurt (considering we're not changing the behavior).